core/src/main/java/hudson/security/AccessDeniedException2.java

```diff
@@ -38,8 +43,10 @@ public AccessDeniedException2(Throwable t, Authentication authentication, Permis
      */
     public void reportAsHeaders(HttpServletResponse rsp) {
         rsp.addHeader("X-You-Are-Authenticated-As",authentication.getName());
-        for (GrantedAuthority auth : authentication.getAuthorities()) {
-            rsp.addHeader("X-You-Are-In-Group",auth.getAuthority());
+        if (REPORT_GROUP_HEADERS) {
```

Comment:
If false, I would rather return a message like "Disabled due to JENKINS-39402; use /whoAmI to diagnose groups. Or set the XXX system property to true" within the Header value